pkg/agent/manager/cache/cache.go

```diff
@@ -708,9 +903,9 @@ func (c *Cache) getSelectorIndexForRead(s selector) *selectorIndex {
 }
 
 type cacheRecord struct {
-	entry *common.RegistrationEntry
-	svid  *X509SVID
-	subs  map[*subscriber]struct{}
+	entry               *common.RegistrationEntry
+	subs                map[*subscriber]struct{}
+	lastAccessTimestamp int64
```

Comment:
If users start getting lot of cache misses then recommendation would be to increase the `X509SVIDCacheMaxSize` instead of introducing complicated logic. The rational being any solution we provide will be at best probabilistic. That's the reason I removed configurable `X509SVIDCacheExpiryPeriod` field from previous revision of this PR.